### DISCUSSION TITLE
How to register event in pug file?

### DISCUSSION THREAD

**Discussion by KirovAI** [webdiscus/pug-plugin#89] [Q&A]
Here pug code:
input#inputFocus(type='text', onblur="checkFocus()", data-len='5')

Pug generate correct html code but not registered events....
![image](https://github.com/webdiscus/pug-plugin/assets/105173528/7ae14587-dcbe-4e37-9535-061959f12c89)
![image](https://github.com/webdiscus/pug-plugin/assets/105173528/23068011-2cf8-4f48-972c-febd9209843b)
Strange

**Comment by webdiscus** [comment]
What is the `TestAV`? 
Can you share completely code of JS file where is defined the `checkFocus()`?

Pug is no framework, you don't need to register in Pug an event. You can write into Pug attributes any JS code as the string, that will be run in a browser.

---
The reference error say what the `checkFocus()` function is not found in your JS code. How do you load the source JS file in Pug?

**Reply by KirovAI** [reply]
I write next simple code for clarity of understanding the issue
pug code(another.html):

```pug
doctype html
html(lang="en")
  head
    meta(charset="UTF-8")
    meta(http-equiv="X-UA-Compatible" content="IE=edge")
    meta(name="viewport" content="width=device-width, initial-scale=1.0")
    title Document
  body
    script(src=require("../js/script.js"))
    h1(onclick="TestAv()") PUG templating engine
    div#root
```
js code(script.js):
```js
function TestAv(){
  alert("work");
}
```
Error:
![image](https://github.com/webdiscus/pug-plugin/assets/105173528/1ed009c3-623a-4816-bab2-4ce118a41cf3)
![image](https://github.com/webdiscus/pug-plugin/assets/105173528/4188db04-22e2-4760-8526-6ff79fa197e2)

I encounter this issue when register in pug any event and trigger it, but other js code executing fine and works. Does it even possible or pug doesnt provide it?
